### PR TITLE
Add join by room id endpoint

### PR DESCRIPTION
**Pull request by exul**
My first attempt to contribute to Ruma :smile:.

I'll need some endpoints for my own project and intend to add them to this library.

I started with a single endpoint but will likely add more. This is mainly to see if I got everything right/what I have to improve. So feedback is appreciated.
For example, I'm not sure about the name of the module, since there are two ways to join a room (by id and by room alias) it cannot be `join_room`.

### CODE REVIEW

File: src/r0/membership.rs [hunk 1]
```diff
@@ -1 +1,63 @@
...
+                "/_matrix/client/r0/rooms/{}/join",
+                params.room_id
+            )
+        }
+
+        fn router_path() -> String {
+            "/_matrix/client/r0/rooms/:room_id/join".to_string()
+        }
+    }
+}
```
Comment (farodin91) [inline src/r0/membership.rs]: add new line

Comment (farodin91) [inline src/r0/membership.rs]: add new line

Comment (exul) [inline src/r0/membership.rs]: Just a new line at the end of the file after the last `}`? I checked the other files, the do not have that.

Comment (exul) [inline src/r0/membership.rs]: done

Comment (farodin91) [inline src/r0/membership.rs]: Thanks

**Comment by jimmycuadra** [comment]
Thanks! :D

**Pull request merged**